### DISCUSSION TITLE
Rust, Go wrappers

### DISCUSSION THREAD

**Discussion by DaveSkender** [DaveSkender/Stock.Indicators#1187] [Dev notes]
> I am also planning to use [the C# DLL] in rust and go, and create a library for rust and go users as well.

_Originally posted by @hmuhdkamran in https://github.com/DaveSkender/Stock.Indicators/issues/1184#issuecomment-2041020725_

> If you get started with [Go](https://go.dev) and/or [Rust](https://www.rust-lang.org), think you can commit to it long term, and want to add it to our collection with some co-branding on _go/rust.stockindicators.dev_, let me know.  I’ll likely be moving these repos out of my personal account over to the @facioquo Org in the next months, and can start adding contributions there.

_Originally reply by @DaveSkender_

 > Let's have a plan for it, and by proper timeline, let’s start on it. I will make spare time for it and some weekends too, and let it start for the finish.

_Originally reply by @hmuhdkamran_

**Comment by DaveSkender** [comment]
@hmuhdkamran it's entirely up to you.  I realize these open-source projects end up being people's 2nd and 3rd priorities; and most don't make it to a full production quality and into a well maintained state.  I'll propose 2 paths for you to consider, to get started:

1. **Option 1**: you create repos in your personal account to get them started.  After some time, if you think it would be something you'd like to continue with, you can transfer them to @facioquo where it'd continue to be developed as an OSS project with more support from our organization and contributors.
2. **Option 2**: we start with a repo in @facioquo that I can setup with some basic scaffolding that we use in all our repos, and then essentially do the same "let's see if it lasts" test on it.  We'd be able to give more feedback, guidance, and support as you get started with this option.

@LeeDongGeon1996, the maintainer of our Python wrapper, went with option 2 and has been dedicated to its success for several years now and is doing an outstanding job keeping it up to date and  with corresponding changes as the base C# project evolves, and addressing user issues as they come up.  He'd be able to give you some insight into what it takes to maintain a language wrapper.

**Reply by LeeDongGeon1996** [reply]
@hmuhdkamran Welcome to be one of our contributors, ~even though it's still under the planning😅~.
As responsible for Python wrapper, I think we've focused on how Python users can have same experience with when they use C# library. We've made it have same interface as much as possible and provide typed interface from the api under the dll. Also, I'm not sure how Rust or Go wrapper will be implemented, since wrapper should have round-trip to .NET runtime, we've also focused on reducing conversion cost (in terms of time or memory). Anyway, it may be just in case of Python wrapper. if there's any better way, you can change or suggest it. I think the most important thing on OSS project is to develop while enjoying.

**Reply by hmuhdkamran** [reply]
Let's start with second option, and hoping for the best to finish it and maintain it, we may build a good team and can acomplish something more than what we are going to start or contribute

**Reply by DaveSkender** [reply]
@hmuhdkamran I've setup [the repo for Go](https://github.com/facioquo/stock-indicators-go) with a base with common docs site setup and added you as maintainer.  Let me know if it's an okay starting point for you.  If so, I'll do the same for a Rust repo.

Also, [connect with me here](https://docs.google.com/forms/d/1ojdLRTN5UzN6OErfz4wHLlvVb6jy91vCV8o6lqs8M0M) so I can get your email address, so i can add you to other sites like Codacy, Slack, etc.

**Reply by hmuhdkamran** [reply]
@DaveSkender I am about to start with the developments in go, so we need a road map to strategize how we are going to approach the product.

**Reply by DaveSkender** [reply]
I've setup [a repo](https://github.com/facioquo/stock-indicators-go) and [a project](https://github.com/orgs/facioquo/projects/15), but you'll need to accept the invitation before I can add you to edit the project.
